src/chainparams.cpp

```diff
@@ -55,6 +56,25 @@ static CBlock CreateGenesisBlock(uint32_t nTime, uint32_t nNonce, uint32_t nBits
     return CreateGenesisBlock(pszTimestamp, genesisOutputScript, nTime, nNonce, nBits, nVersion, genesisReward);
 }
 
+/**
+ * Helper for initializing vDeployments via util::ArrayInit
+ */
+static constexpr util::NamedArgument<int, struct bit_arg> _bit;
+static constexpr util::NamedArgument<int64_t, struct start_time_arg> _start_time;
+static constexpr util::NamedArgument<int64_t, struct timeout_arg> _timeout;
+static constexpr util::NamedArgument<int, struct minactheight_arg> _minactheight;
+
+template<size_t O>
+static inline util::ArrayInitElement<O, Consensus::BIP9Deployment> Dep(decltype(_bit)::Arg bit, decltype(_start_time)::Arg start_time, decltype(_timeout)::Arg timeout, decltype(_minactheight)::Arg minactheight)
+{
+    Consensus::BIP9Deployment dep;
+    dep.bit = bit.v;
+    dep.nStartTime = start_time.v;
+    dep.nTimeout = timeout.v;
+    dep.min_activation_height = minactheight.v;
+    return util::ArrayInitElement<O, Consensus::BIP9Deployment>{dep};
```

Comment:
Can you add a constructor to BIP9Deployment that takes these values? That way this method body collapses to just 1 line, and we'd then be passing a temporary

Reply:
gcc and clang both optimise them to the same thing, and saying `dep.bit = bit.v` etc seems a bit more obvious that everythings in the right order than `BIP9Deployment(bit.v, start_time.v, timeout.v, minactheight.v)` does. (per [godbolt](https://godbolt.org/z/Th4cc7j9f), VS seems to generate some moderately weird assembly, but changing how the temporary BIP9Deployment struct is created initialised doesn't seem to improve things)